src/components/input/StateEditor.ts

```diff
@@ -0,0 +1,72 @@
+import { State } from "./State"
+
+export class StateEditor {
+	get value(): string { return this.state.value }
+	private get selectionStart(): number { return this.state.selectionStart }
+	private set selectionStart(selectionStart: number) { this.state.selectionStart = selectionStart }
+	private get selectionEnd(): number { return this.state.selectionEnd }
+	private set selectionEnd(selectionEnd: number) { this.state.selectionEnd = selectionEnd }
+	get stateCopy(): State { return { value: this.state.value, selectionStart: this.selectionStart, selectionEnd: this.selectionEnd}}
```

Comment:
We need a better name, let's discuss options. Probably rename the current `state` to `backend` or ` data` and let this property be named `state`. Another alternative would be to have a method instead called something like `asState()`, 'toState()`.

Reply:
Given these options I'm more in favor of the "toState()"-solution. The term "data" is very vague and "backend" doesn't sound very accurate to me. I'll go with the "toState()"-solution for now if you don't mind.

Reply:
The name of a private property does not have to be perfect as it will only be read within that context. However do a `toState()` method as that makes it clearer that it will be a copy and not the internal state.